release 2024-03-19 (#1760)
tigerbeetle/tigerbeetle: CHANGELOG.md

# TigerBeetle Changelog

## 2024-03-19

- Bump version to 0.15.x

> - Starting with 0.15.x, TigerBeetle is ready for production use, guarantees to never lose data, and
>   provides forward upgrade path through storage stability.

Reviewer: ```suggestion
- Starting with 0.15.x, TigerBeetle is ready for production use, preserves durability and
  provides a forward upgrade path through storage stability.
```
